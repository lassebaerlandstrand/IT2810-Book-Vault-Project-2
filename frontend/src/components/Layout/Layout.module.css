.wrapper {
  margin: 0 auto;
  width: 100%;
  max-width: $mantine-breakpoint-md;
}
