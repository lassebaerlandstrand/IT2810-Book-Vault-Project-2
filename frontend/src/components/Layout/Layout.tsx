import { Outlet } from 'react-router-dom';
import { AppShell } from '@mantine/core';
import styles from './Layout.module.css';

const Layout = () => {
  return (
    <AppShell padding="md" header={{ height: 50 }} /* Number is automatically converted to rem */>
      <AppShell.Header>TODO: Header component here</AppShell.Header>
      <AppShell.Main className={styles.wrapper}>
        <Outlet />
      </AppShell.Main>
    </AppShell>
  );
};

export default Layout;
