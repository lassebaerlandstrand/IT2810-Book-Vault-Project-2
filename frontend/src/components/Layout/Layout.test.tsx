import { render } from '@test-utils';
import Layout from './Layout';

describe('Layout', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Layout />);
    expect(asFragment()).toMatchSnapshot();
  });
});
