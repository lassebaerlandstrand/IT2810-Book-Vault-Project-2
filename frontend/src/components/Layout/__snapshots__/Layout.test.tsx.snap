// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-font-family-headings: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-breakpoint-xs: 30em;--mantine-breakpoint-md: 64em;--mantine-breakpoint-xl: 90em;--mantine-breakpoint-xxs: 24em;--mantine-color-logo-theme-0: #fefefe;--mantine-color-logo-theme-1: #cbcfd2;--mantine-color-logo-theme-2: #97a0a6;--mantine-color-logo-theme-3: #7d8991;--mantine-color-logo-theme-4: #63717b;--mantine-color-logo-theme-5: #495a65;--mantine-color-logo-theme-6: #2f434f;--mantine-color-logo-theme-7: #132937;--mantine-color-logo-theme-8: #0b202e;--mantine-color-logo-theme-9: #061017;}:root[data-mantine-color-scheme="dark"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-8);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-9);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.15);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.2);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-3);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-outline-hover: rgba(99, 113, 123, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-filled);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-7);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.1);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.12);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline-hover: rgba(47, 67, 79, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 29.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 30em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 63.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 64em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 89.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 90em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 23.99375em) {.mantine-visible-from-xxs {display: none !important;}}@media (min-width: 24em) {.mantine-hidden-from-xxs {display: none !important;}}
  </style>
  <style
    data-mantine-styles="inline"
  >
    :root{--app-shell-header-height:calc(3.5rem * var(--mantine-scale));--app-shell-header-offset:calc(3.5rem * var(--mantine-scale));--app-shell-padding:var(--mantine-spacing-xl);}
  </style>
  <div
    class="m_89ab340 mantine-AppShell-root"
    data-resizing="true"
    style="--app-shell-transition-duration: 200ms; --app-shell-transition-timing-function: ease; width: calc(100vw - 0px);"
  >
    <header
      class="m_3b16f56b mantine-AppShell-header right-scroll-bar-position _header_0c8bf9"
      data-with-border="true"
      style="width: 100vw; padding-right: 0px; --app-shell-header-z-index: 100;"
    >
      <div
        class="_wrapper_c634d8 m_7485cace mantine-Container-root"
      >
        <a
          class="_logo_c634d8"
          href="/"
        >
          <figure
            class="_wrapper_16f863  m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <img
              class="m_9e117634 mantine-Image-root"
              src="/project2/src/assets/BookVaultLogo.png"
            />
            <svg
              height="0.326667in"
              viewBox="0 0 468 98"
              width="1.56in"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 75.00,2.00            C 75.00,2.00 75.00,4.86 75.00,4.86              75.00,4.86 65.34,4.86 65.34,4.86              65.34,4.86 65.34,35.00 65.34,35.00              65.34,35.00 62.00,35.00 62.00,35.00              62.00,35.00 62.00,4.86 62.00,4.86              62.00,4.86 53.00,4.86 53.00,4.86              53.00,4.86 53.00,2.00 53.00,2.00              53.00,2.00 75.00,2.00 75.00,2.00 Z            M 92.20,9.00            C 94.12,9.00 95.85,9.41 97.39,10.22              98.93,11.02 100.14,12.26 101.02,13.92              101.90,15.59 102.34,17.62 102.34,20.03              102.34,20.03 102.34,35.00 102.34,35.00              102.34,35.00 99.00,35.00 99.00,35.00              99.00,35.00 99.00,20.42 99.00,20.42              99.00,17.64 98.31,15.52 96.94,14.06              95.57,12.59 93.71,11.86 91.34,11.86              88.91,11.86 86.96,12.64 85.52,14.20              84.07,15.76 83.34,18.05 83.34,21.08              83.34,21.08 83.34,35.00 83.34,35.00              83.34,35.00 80.00,35.00 80.00,35.00              80.00,35.00 80.00,0.00 80.00,0.00              80.00,0.00 83.34,0.00 83.34,0.00              83.34,0.00 83.34,13.83 83.34,13.83              84.14,12.30 85.33,11.11 86.91,10.27              88.48,9.42 90.24,9.00 92.20,9.00 Z            M 133.00,20.50            C 133.00,21.59 132.97,22.43 132.91,23.00              132.91,23.00 111.34,23.00 111.34,23.00              111.34,24.94 111.77,26.59 112.64,27.97              113.50,29.34 114.64,30.39 116.05,31.09              117.46,31.79 119.00,32.14 120.67,32.14              122.85,32.14 124.68,31.62 126.17,30.59              127.66,29.56 128.64,28.17 129.12,26.42              129.12,26.42 132.62,26.42 132.62,26.42              131.98,28.95 130.62,31.02 128.55,32.61              126.47,34.20 123.88,35.00 120.75,35.00              118.30,35.00 116.11,34.47 114.17,33.41              112.24,32.34 110.73,30.84 109.64,28.88              108.55,26.91 108.00,24.62 108.00,22.00              108.00,19.38 108.54,17.08 109.61,15.11              110.69,13.13 112.20,11.62 114.12,10.58              116.06,9.53 118.27,9.00 120.75,9.00              123.21,9.00 125.36,9.52 127.22,10.55              129.07,11.58 130.50,12.96 131.50,14.70              132.50,16.45 133.00,18.38 133.00,20.50 Z            M 129.66,20.14            C 129.66,18.37 129.24,16.87 128.42,15.62              127.61,14.39 126.49,13.45 125.06,12.81              123.64,12.18 122.10,11.86 120.44,11.86              117.91,11.86 115.77,12.59 114.00,14.03              112.23,15.47 111.34,17.51 111.34,20.14              111.34,20.14 129.66,20.14 129.66,20.14 Z            M 32.83,65.38            C 36.22,65.97 39.10,67.71 41.45,70.61              43.82,73.50 45.00,76.77 45.00,80.44              45.00,83.56 44.18,86.38 42.55,88.89              40.92,91.40 38.57,93.38 35.48,94.83              32.41,96.28 28.83,97.00 24.73,97.00              24.73,97.00 0.00,97.00 0.00,97.00              0.00,97.00 0.00,35.00 0.00,35.00              0.00,35.00 23.44,35.00 23.44,35.00              27.62,35.00 31.22,35.72 34.22,37.16              37.23,38.59 39.50,40.53 41.03,42.95              42.56,45.38 43.33,48.09 43.33,51.09              43.33,54.68 42.38,57.67 40.48,60.08              38.59,62.48 36.04,64.24 32.83,65.38 Z            M 10.33,61.00            C 10.33,61.00 22.62,61.00 22.62,61.00              25.88,61.00 28.41,60.27 30.25,58.81              32.08,57.34 33.00,55.23 33.00,52.47              33.00,49.77 32.08,47.66 30.25,46.14              28.41,44.61 25.88,43.84 22.62,43.84              22.62,43.84 10.33,43.84 10.33,43.84              10.33,43.84 10.33,61.00 10.33,61.00 Z            M 23.81,88.16            C 27.20,88.16 29.85,87.37 31.78,85.80              33.71,84.23 34.67,82.03 34.67,79.20              34.67,76.34 33.64,74.06 31.59,72.38              29.55,70.69 26.84,69.84 23.47,69.84              23.47,69.84 10.33,69.84 10.33,69.84              10.33,69.84 10.33,88.16 10.33,88.16              10.33,88.16 23.81,88.16 23.81,88.16 Z            M 76.56,97.50            C 71.93,97.50 67.74,96.45 64.02,94.36              60.29,92.26 57.35,89.30 55.20,85.48              53.07,81.67 52.00,77.25 52.00,72.22              52.00,67.24 53.10,62.83 55.30,58.98              57.49,55.14 60.49,52.18 64.28,50.11              68.07,48.04 72.31,47.00 77.00,47.00              81.69,47.00 85.93,48.04 89.72,50.11              93.51,52.18 96.50,55.14 98.70,58.98              100.90,62.83 102.00,67.24 102.00,72.22              102.00,77.19 100.87,81.59 98.61,85.44              96.36,89.28 93.29,92.26 89.41,94.36              85.53,96.45 81.25,97.50 76.56,97.50 Z            M 76.61,88.66            C 79.21,88.66 81.66,88.04 83.95,86.80              86.24,85.56 88.10,83.70 89.53,81.23              90.96,78.76 91.67,75.75 91.67,72.20              91.67,68.66 90.98,65.67 89.61,63.23              88.24,60.79 86.43,58.95 84.17,57.70              81.92,56.46 79.49,55.84 76.88,55.84              74.25,55.84 71.82,56.46 69.59,57.70              67.38,58.95 65.61,60.79 64.30,63.23              62.98,65.67 62.33,68.66 62.33,72.20              62.33,77.45 63.68,81.51 66.39,84.38              69.10,87.23 72.50,88.66 76.61,88.66 Z            M 133.56,97.50            C 128.93,97.50 124.74,96.45 121.02,94.36              117.29,92.26 114.35,89.30 112.20,85.48              110.07,81.67 109.00,77.25 109.00,72.22              109.00,67.24 110.10,62.83 112.30,58.98              114.49,55.14 117.49,52.18 121.28,50.11              125.07,48.04 129.31,47.00 134.00,47.00              138.69,47.00 142.93,48.04 146.72,50.11              150.51,52.18 153.50,55.14 155.70,58.98              157.90,62.83 159.00,67.24 159.00,72.22              159.00,77.19 157.87,81.59 155.61,85.44              153.36,89.28 150.29,92.26 146.41,94.36              142.53,96.45 138.25,97.50 133.56,97.50 Z            M 133.61,88.66            C 136.21,88.66 138.66,88.04 140.95,86.80              143.24,85.56 145.10,83.70 146.53,81.23              147.96,78.76 148.67,75.75 148.67,72.20              148.67,68.66 147.98,65.67 146.61,63.23              145.24,60.79 143.43,58.95 141.17,57.70              138.92,56.46 136.49,55.84 133.88,55.84              131.25,55.84 128.82,56.46 126.59,57.70              124.38,58.95 122.61,60.79 121.30,63.23              119.98,65.67 119.33,68.66 119.33,72.20              119.33,77.45 120.68,81.51 123.39,84.38              126.10,87.23 129.50,88.66 133.61,88.66 Z            M 189.58,72.55            C 189.58,72.55 212.19,97.00 212.19,97.00              212.19,97.00 198.48,97.00 198.48,97.00              198.48,97.00 180.33,75.92 180.33,75.92              180.33,75.92 180.33,97.00 180.33,97.00              180.33,97.00 170.00,97.00 170.00,97.00              170.00,97.00 170.00,31.00 170.00,31.00              170.00,31.00 180.33,31.00 180.33,31.00              180.33,31.00 180.33,69.44 180.33,69.44              180.33,69.44 198.12,48.00 198.12,48.00              198.12,48.00 212.19,48.00 212.19,48.00              212.19,48.00 189.58,72.55 189.58,72.55 Z            M 294.98,35.00            C 294.98,35.00 271.75,97.00 271.75,97.00              271.75,97.00 260.02,97.00 260.02,97.00              260.02,97.00 236.69,35.00 236.69,35.00              236.69,35.00 247.55,35.00 247.55,35.00              247.55,35.00 265.89,86.56 265.89,86.56              265.89,86.56 284.31,35.00 284.31,35.00              284.31,35.00 294.98,35.00 294.98,35.00 Z            M 300.00,72.05            C 300.00,67.12 301.02,62.77 303.06,58.98              305.10,55.20 307.88,52.26 311.39,50.16              314.91,48.05 318.80,47.00 323.06,47.00              326.91,47.00 330.26,47.77 333.12,49.30              335.99,50.82 338.28,52.71 340.00,54.97              340.00,54.97 340.00,48.00 340.00,48.00              340.00,48.00 350.33,48.00 350.33,48.00              350.33,48.00 350.33,97.00 350.33,97.00              350.33,97.00 340.00,97.00 340.00,97.00              340.00,97.00 340.00,89.39 340.00,89.39              338.28,91.71 335.94,93.65 332.98,95.19              330.04,96.73 326.66,97.50 322.88,97.50              318.68,97.50 314.83,96.42 311.34,94.27              307.86,92.10 305.10,89.09 303.06,85.22              301.02,81.34 300.00,76.95 300.00,72.05 Z            M 340.00,72.20            C 340.00,68.84 339.30,65.92 337.91,63.45              336.52,60.98 334.70,59.09 332.44,57.80              330.19,56.50 327.76,55.84 325.16,55.84              322.56,55.84 320.13,56.48 317.88,57.75              315.62,59.02 313.80,60.88 312.41,63.33              311.02,65.77 310.33,68.67 310.33,72.03              310.33,75.39 311.02,78.34 312.41,80.88              313.80,83.41 315.64,85.34 317.92,86.67              320.20,88.00 322.61,88.66 325.16,88.66              327.76,88.66 330.19,88.01 332.44,86.70              334.70,85.40 336.52,83.50 337.91,81.00              339.30,78.50 340.00,75.57 340.00,72.20 Z            M 408.00,48.00            C 408.00,48.00 408.00,97.00 408.00,97.00              408.00,97.00 397.67,97.00 397.67,97.00              397.67,97.00 397.67,91.05 397.67,91.05              396.07,93.05 393.98,94.62 391.41,95.78              388.83,96.93 386.09,97.50 383.19,97.50              379.32,97.50 375.86,96.70 372.81,95.11              369.77,93.52 367.38,91.15 365.62,88.02              363.88,84.88 363.00,81.11 363.00,76.69              363.00,76.69 363.00,48.00 363.00,48.00              363.00,48.00 373.33,48.00 373.33,48.00              373.33,48.00 373.33,75.14 373.33,75.14              373.33,79.50 374.41,82.84 376.58,85.17              378.75,87.50 381.71,88.66 385.45,88.66              389.20,88.66 392.18,87.50 394.38,85.17              396.57,82.84 397.67,79.50 397.67,75.14              397.67,75.14 397.67,48.00 397.67,48.00              397.67,48.00 408.00,48.00 408.00,48.00 Z            M 432.33,31.00            C 432.33,31.00 432.33,97.00 432.33,97.00              432.33,97.00 422.00,97.00 422.00,97.00              422.00,97.00 422.00,31.00 422.00,31.00              422.00,31.00 432.33,31.00 432.33,31.00 Z            M 456.00,56.00            C 456.00,56.00 456.00,83.03 456.00,83.03              456.00,84.86 456.43,86.17 457.30,86.97              458.16,87.76 459.63,88.16 461.72,88.16              461.72,88.16 468.00,88.16 468.00,88.16              468.00,88.16 468.00,97.00 468.00,97.00              468.00,97.00 459.92,97.00 459.92,97.00              455.33,97.00 451.80,95.93 449.34,93.78              446.89,91.64 445.67,88.15 445.67,83.31              445.67,83.31 445.67,56.00 445.67,56.00              445.67,56.00 440.00,56.00 440.00,56.00              440.00,56.00 440.00,47.16 440.00,47.16              440.00,47.16 445.67,47.16 445.67,47.16              445.67,47.16 445.67,35.50 445.67,35.50              445.67,35.50 456.00,35.50 456.00,35.50              456.00,35.50 456.00,47.16 456.00,47.16              456.00,47.16 468.00,47.16 468.00,47.16              468.00,47.16 468.00,56.00 468.00,56.00              468.00,56.00 456.00,56.00 456.00,56.00 Z"
                fill="currentColor"
                id="TextLogo"
              />
            </svg>
          </figure>
        </a>
        <div
          class="_linksDesktop_c634d8 m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-sm); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <a
            class="_link_c634d8"
            href="/"
          >
            HOME
          </a>
          <a
            class="_link_c634d8"
            href="/books"
          >
            BOOKS
          </a>
          <button
            aria-label="Change color theme"
            class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
            data-size="lg"
            data-variant="default"
            style="--ai-size: var(--ai-size-lg); --ai-bg: var(--mantine-color-default); --ai-hover: var(--mantine-color-default-hover); --ai-color: var(--mantine-color-default-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
            type="button"
          >
            <span
              class="m_8d3afb97 mantine-ActionIcon-icon"
            >
              <svg
                class="tabler-icon tabler-icon-moon "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
              </svg>
            </span>
          </button>
        </div>
        <button
          class="mantine-focus-auto _burgerIcon_c634d8 m_fea6bf1a mantine-Burger-root m_87cf2631 mantine-UnstyledButton-root"
          type="button"
        >
          <div
            class="m_d4fb9cad mantine-Burger-burger"
            data-reduce-motion="true"
          />
        </button>
      </div>
    </header>
    <main
      class="m_8983817 mantine-AppShell-main _wrapper_0c8bf9"
    />
  </div>
</DocumentFragment>
`;
